### PR TITLE
Replace mocks with test implementation in RepositoriesServiceTests

### PR DESCRIPTION
**Pull request by mhl-b**
I observed that `testRegisterRepositorySuccessAfterCreationFailed` test never invokes assertion blocks, because listener is not invoked.

There are 2 problems:

1. Test setup used mocks. Mocks interrupt listener chain propagation, so registerRepository never returned Response or Failure.
2. We silently ignore assertions in listener because it is not invoked. Test pass successfully.

PutRepositories method relies on cluster state update. I replace mocked ClusterService and ThreadPool with test implementation of these. Also add blocking call on listener to ensure we get result.

Address [comment](https://github.com/elastic/elasticsearch/pull/108531#pullrequestreview-2051318976) to break down larger PR into smaller pieces in #108531

### CODE REVIEW

**Review by idegtiarenko — APPROVED** [review]

File: server/src/test/java/org/elasticsearch/repositories/RepositoriesServiceTests.java [hunk 1]
```diff
@@ -82,10 +96,18 @@ public void setUp() throws Exception {
...
         );
-        final ClusterApplierService clusterApplierService = mock(ClusterApplierService.class);
-        when(clusterApplierService.threadPool()).thenReturn(threadPool);
-        final ClusterService clusterService = mock(ClusterService.class);
-        when(clusterService.getClusterApplierService()).thenReturn(clusterApplierService);
+
+        clusterService = ClusterServiceUtils.createClusterService(threadPool);
+
+        // cluster utils publisher does not call AckListener, making some method calls hang indefinitely
+        // in this test we have a single master node, and it acknowledges cluster state immediately
```
Comment (DaveCTurner) [inline server/src/test/java/org/elasticsearch/repositories/RepositoriesServiceTests.java:103]: :+1:

File: server/src/test/java/org/elasticsearch/repositories/RepositoriesServiceTests.java [hunk 2]
```diff
@@ -283,18 +314,11 @@ public void testRegisterRepositorySuccessAfterCreationFailed() {
...
-
-            @Override
-            public void onFailure(Exception e) {
-                assert false : e;
-            }
-        });
+        var resultListener = new SubscribableListener<AcknowledgedResponse>();
+        repositoriesService.registerRepository(request, resultListener);
+        var resp = safeAwait(resultListener);
+        assertTrue(resp.isAcknowledged());
```
Comment (DaveCTurner) [inline server/src/test/java/org/elasticsearch/repositories/RepositoriesServiceTests.java:320]: nit: would rather not use abbrs. for var. names wherever poss. :)

```suggestion
        var response = safeAwait(resultListener);
        assertTrue(response.isAcknowledged());
```

**Review by DaveCTurner — APPROVED** [review]
LGTM2, one nit

File: server/src/test/java/org/elasticsearch/repositories/RepositoriesServiceTests.java (hunk 2 above)
Comment (mhl-b) [reply server/src/test/java/org/elasticsearch/repositories/RepositoriesServiceTests.java:320]: ofc

**Pull request merged**